### DISCUSSION TITLE
What is an icon class?

### DISCUSSION THREAD

**Discussion by Tony20221** [oqtane/oqtane.framework#3162] [Q&A]
When adding a new page, there's an option to specify an icon. An icon class. What does it look like?
Is there documentation that explains all the settings?

I suggest putting an example in the help tooltip or as a placeholder in the textbox so users have an idea what it looks like.
At least if it shows an example, I can do a search in the files and know what to put in there.

**Comment by vnetonline** [comment]
Look at the home page it has the home icon in there it is configured using `oi oi-home`

**Reply by Tony20221** [reply]
Thanks. However, I find this to be an issue because of no discoverability. Is Oqtane making an assumption that all users who create pages are technical people and will figure out what kind of icon Oqtane is asking for?

I had to search and find out that oi-home is in open-iconic-bootstrap.min.css, unminify it to figure out what other icons are in there.
I think this is putting too much burden on the user including technical users.
Instead of the textbox, this can be a dropdown with existing icons, showing the icons along with the name, where the user can just select one plus a textbox at the top so a user can select an icon which is not one from the standard ones. Upon selection, the proper text like 'oi oi-home', will be inserted. 
Or at least put a hyperlink in the tooltip to a page that shows the icons from open-iconic with their text represnetation.

**Reply by leigh-pointer** [reply]
I somewhat agree.  I have a solution and will create a PR for this.

**Comment by leigh-pointer** [comment]
PR created #3167

**Reply by vnetonline** [reply]
Good work on this PR @leigh-pointer

**Comment by sbwalker** [comment]
@Tony20221 "Is Oqtane making an assumption that all users who create pages are technical people" - the answer to that question is Yes - it is primarily an application framework and the main audience is developers. 

@leigh-pointer note that Oqtane does not only support the open iconic icon library - it supports other icon libraries as well.

**Reply by leigh-pointer** [reply]
@sbwalker the PR control allows edit! I tested with my theme and the icon resources. Rather than not knowing what enter it does give the choice from the already available resource.

**Reply by sbwalker** [reply]
@leigh-pointer ok I will review - thank you

**Comment by sbwalker** [comment]
@Tony20221 a more user friendly solution was added for Icons by @leigh-pointer - it will be included in the next release (4.0.3). The solution includes an InputList component which developers will be able to use in their own custom modules for allowing users to enter data based on list of options.
